experimental/components/VideoPlayer/demo/src/jvmMain/kotlin/org/jetbrains/compose/videoplayer/demo/Main.kt

```diff
@@ -1,22 +1,94 @@
 package org.jetbrains.compose.videoplayer.demo
 
-import androidx.compose.material.MaterialTheme
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.unit.dp
-import androidx.compose.ui.window.WindowState
 import androidx.compose.ui.window.singleWindowApplication
 import org.jetbrains.compose.videoplayer.VideoPlayer
 
 fun main() {
-    singleWindowApplication(
-        title = "Video Player",
-        state = WindowState(width = 800.dp, height = 800.dp)
-    ) {
+    singleWindowApplication(title = "Video Player") {
         MaterialTheme {
-            VideoPlayer(
-                url = "http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4",
-                width = 640,
-                height = 480
+            App()
+        }
+    }
+}
+
+@Composable
+fun App() {
+    var seek by remember { mutableStateOf(0f) }
+    var speed by remember { mutableStateOf(1f) }
+    var volume by remember { mutableStateOf(1f) }
+    var isResumed by remember { mutableStateOf(true) }
+    var isFullscreen by remember { mutableStateOf(false) }
+    val stopPlayback = remember { { isResumed = false } }
+    val toggleResume = remember { { isResumed = !isResumed } }
+    val toggleFullscreen = remember { { isFullscreen = !isFullscreen } }
+    Column {
+        val progress by VideoPlayer(
+            url = "http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4",
+            seek = seek,
+            speed = speed,
+            volume = volume,
+            isResumed = isResumed,
+            isFullscreen = isFullscreen,
+            onFinish = stopPlayback,
+            modifier = Modifier
+                .fillMaxWidth()
+                .height(400.dp)
+        )
```

Comment:
How about to create a `VideoPlayerState` and a `rememberVideoPlayerState()` like it is done with `rememberScrollState()` and a bunch of Compose APIs.

Another API sample: https://github.com/google/accompanist/blob/main/web/src/main/java/com/google/accompanist/web/WebView.kt#L311

Reply:
Looks like it's resolved :)